src/main/java/storage/Storage.java

```diff
@@ -87,30 +95,33 @@ public static Storage getInstance() {
     public AuthorList readAuthorListFromDataFile() {
         assert dataFile != null : "dataFile cannot be null";
         try (FileReader reader = new FileReader(dataFile)) {
-            /*
-            The weird syntax in the following line is required because of type erasure. It allows us to capture the
-            generic type information at runtime before it gets erased.
-             */
-            Type authorListType = (new TypeToken<AuthorList>() {}).getType();
-            AuthorList authorList = gson.fromJson(reader, authorListType);
+            AuthorList authorList = gson.fromJson(reader, AUTHOR_LIST_TYPE);
             logger.info("Data restored");
-            System.out.println("Successfully restored data!");
+            System.out.println("Data restored!");
             return authorList;
         } catch (IOException e) {
-            logger.warning("Problems accessing file, data was not restored!" + e.getMessage());
-            System.out.println("Problems accessing file, data was not restored! Continuing with an empty list...");
+            logger.warning("Problems accessing file: " + e.getMessage());
+            System.out.println("Problems accessing file, data was not restored! Continuing with an empty list.");
+        } catch (JsonSyntaxException e) {
+            logger.warning("JSON from file is malformed: " + e.getMessage());
+            System.out.println(
+                    "JSON from file is malformed, data was not restored! Continuing with an empty list."
```

Comment:
Good job catching the possibility of a corrupted file and handling the case for it!